muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/rest/CommunicatorRecipientsRESTService.java

```diff
@@ -0,0 +1,308 @@
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.commons.lang3.StringUtils;
+
+import fi.otavanopisto.muikku.model.users.EnvironmentRoleArchetype;
+import fi.otavanopisto.muikku.model.users.EnvironmentUser;
+import fi.otavanopisto.muikku.model.users.UserEntity;
+import fi.otavanopisto.muikku.model.users.UserGroupEntity;
+import fi.otavanopisto.muikku.model.workspace.WorkspaceAccess;
+import fi.otavanopisto.muikku.model.workspace.WorkspaceEntity;
+import fi.otavanopisto.muikku.plugin.PluginRESTService;
+import fi.otavanopisto.muikku.plugins.communicator.CommunicatorPermissionCollection;
+import fi.otavanopisto.muikku.plugins.workspace.rest.model.WorkspaceBasicInfo;
+import fi.otavanopisto.muikku.rest.RESTPermitUnimplemented;
+import fi.otavanopisto.muikku.schooldata.RestCatchSchoolDataExceptions;
+import fi.otavanopisto.muikku.schooldata.SchoolDataBridgeSessionController;
+import fi.otavanopisto.muikku.schooldata.SchoolDataIdentifier;
+import fi.otavanopisto.muikku.schooldata.WorkspaceEntityController;
+import fi.otavanopisto.muikku.search.SearchProvider;
+import fi.otavanopisto.muikku.search.SearchProvider.Sort;
+import fi.otavanopisto.muikku.search.SearchResult;
+import fi.otavanopisto.muikku.security.RoleFeatures;
+import fi.otavanopisto.muikku.session.SessionController;
+import fi.otavanopisto.muikku.users.EnvironmentUserController;
+import fi.otavanopisto.muikku.users.UserEmailEntityController;
+import fi.otavanopisto.muikku.users.UserEntityController;
+import fi.otavanopisto.muikku.users.UserEntityFileController;
+import fi.otavanopisto.muikku.users.UserGroupEntityController;
+import fi.otavanopisto.muikku.users.WorkspaceUserEntityController;
+
+@Path("/communicator")
+@RequestScoped
+@Stateful
+@Produces ("application/json")
+@RestCatchSchoolDataExceptions
+public class CommunicatorRecipientsRESTService extends PluginRESTService {
+
+  private static final long serialVersionUID = 4522150023024829380L;
+
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private SessionController sessionController;
+  
+  @Inject
+  private EnvironmentUserController environmentUserController;
+  
+  @Inject
+  private UserEntityController userEntityController;
+
+  @Inject
+  private UserEntityFileController userEntityFileController;
+
+  @Inject
+  private UserEmailEntityController userEmailEntityController;
+  
+  @Inject
+  private UserGroupEntityController userGroupEntityController;
+
+  @Inject
+  private SchoolDataBridgeSessionController schoolDataBridgeSessionController;
+
+  @Inject
+  private WorkspaceEntityController workspaceEntityController;
+
+  @Inject
+  private WorkspaceUserEntityController workspaceUserEntityController;
+  
+  @Inject
+  @Any
+  private Instance<SearchProvider> searchProviders;
+
+  @GET
+  @Path("/recipientsUsersSearch") // TODO mApi requires ids between all resources - this should be /recipients/users/search
+  @RESTPermitUnimplemented
```

Comment:
Might be more accurate as...
`@RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)`
...as it also does away with the logged in check at the beginning of the method